log/DESIGN.md

```diff
@@ -0,0 +1,234 @@
+# Logs Bridge API
+
+OpenTelemetry Logs tracking issue at [#4696](https://github.com/open-telemetry/opentelemetry-go/issues/4696).
+
+## Abstract
+
+We propose adding a `go.opentelemetry.io/otel/log` Go module which will provide
+[Logs Bridge API](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/).
+
+## Background
+
+The key challenge is to create a well-performant API compliant with the specification.
+Performance is seen as one of the most important characteristics of logging libraries in Go.
+
+## Design
+
+This proposed design aims to:
+
+- be specification compliant,
+- have similar API to Trace and Metrics API,
+- take advantage of both OpenTelemetry and `slog` experience to achieve acceptable performance.
+
+### Module structure
+
+The Go module consits of the following packages:
+
+- `go.opentelemetry.io/otel/log`
+- `go.opentelemetry.io/otel/log/embedded`
+- `go.opentelemetry.io/otel/log/noop`
+
+### LoggerProvider
+
+The [`LoggerProvider` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#loggerprovider)
+is defined as an interface [provider.go](provider.go).
+
+### Logger
+
+The [`Logger` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#logger)
+is defined as an interface in [logger.go](logger.go).
+
+### Record
+
+The [`LogRecord` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#logger)
+is defined as a struct in [record.go](record.go).
+
+`Record` has `Attributes` and `AddAttributes` methods,
+like [`slog.Record.Attrs`](https://pkg.go.dev/log/slog#Record.Attrs)
+and [`slog.Record.AddAttrs`](https://pkg.go.dev/log/slog#Record.AddAttrs),
+in order to achieve high-performance when accessing and setting attributes efficiently.
+
+`Record` has a `AttributesLen` method that returns
+the number of attributes to allow slice preallocation
+when converting records to a different representation.
+
+### Usage Example: Log Bridge implementation
+
+Excerpt of a [slog.Handler](https://pkg.go.dev/log/slog#Handler)
+naive implementation.
+
+```go
+type handler struct {
+	logger log.Logger
+}
+
+func (h *handler) Handle(ctx context.Context, r slog.Record) error {
+	lvl := convertLevel(r.Level)
+
+	record := Record{Timestamp: r.Time, Severity: lvl, Body: r.Message}
+
+	if r.AttributesLen() > 5 {
```

Comment:
could we maybe add a comment here / constant out the magic number 5 as to why this optimization works this way? 
`@jba` s talk at `golab` was illuminating on how/why `slog` works like it does: [Structured logging for the standard library](https://golab.io/talks/structured-logging-for-the-standard-library)

Reply:
Thanks. I also plan to add a naïve implementations of `logr` and `slog` bridges to the benchmarks so maybe I will even decide to reference them instead.

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/4725/commits/ca5bfb780c369d70f913107ca870fdff5e4070bb

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/4725/commits/0d9c3353e35ebe7d3e3ebc1332ef93011c4ccf6a